WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsFragment.kt

```diff
@@ -24,33 +28,78 @@ class ReaderInterestsFragment : Fragment(R.layout.reader_interests_fragment_layo
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        initDoneButton()
         initViewModel()
     }
 
+    private fun initDoneButton() {
+        done_button.setOnClickListener {
+            viewModel.onDoneButtonClick()
+        }
+    }
+
     private fun initViewModel() {
-        viewModel = ViewModelProviders.of(this, viewModelFactory).get(ReaderInterestsViewModel::class.java)
+        viewModel = ViewModelProviders.of(this, viewModelFactory)
+            .get(ReaderInterestsViewModel::class.java)
         startObserving()
     }
 
     private fun startObserving() {
         viewModel.uiState.observe(viewLifecycleOwner, Observer { uiState ->
             updateInterests(uiState.interestsUiState)
+            updateDoneButton(uiState.doneButtonUiState)
+        })
+
+        viewModel.navigateToDiscover.observe(viewLifecycleOwner, Observer { event ->
+            event?.getContentIfNotHandled()?.let {
+                navigateToDiscover()
+            }
         })
 
         viewModel.start()
     }
 
+    private fun updateDoneButton(doneButtonUiState: DoneButtonUiState) {
+        with(done_button) {
+            isEnabled = doneButtonUiState.enabled
+            text = getString(doneButtonUiState.titleRes)
+        }
+        uiHelpers.updateVisibility(done_button, doneButtonUiState.visible)
+    }
+
     private fun updateInterests(interestsUiState: List<InterestUiState>) {
         interestsUiState.forEachIndexed { index, interestTagUiState ->
-            val chip = layoutInflater.inflate(R.layout.reader_interest_filter_chip, interests_chip_group, false) as Chip
+            var chip = interests_chip_group.findViewWithTag<Chip>(interestTagUiState.title)
+            if (chip == null) {
```

Comment:
We might want to consider moving the body of this `if` into a separate method, wdyt?

We could write something like 
`val chip = interests_chip_group.findViewWithTag<Chip>(interestTagUiState.title) ?: createChipView(...)`

Reply:
Refactored here: 36dab46